src/coreclr/tools/ILVerification/AccessVerificationHelpers.cs

```diff
@@ -204,11 +187,21 @@ private static bool CanAccessMethodSignature(this TypeDesc currentType, MethodDe
                     return false;
             }
 
-            return currentType.CanAccessSignatureArguments(methodSig);
+            return currentType.CanAccessSignature(methodSig);
```

Comment:
I would delete `CanAccessMethodSignature` and instate changed the one place where it is called to just call `CanAccessSignature(targetMethod.Signature)`.

Reply:
Deleted CanAccessMethodSignature and simplified the checks in CanAccessSignature.